packages/flutter/lib/src/material/stepper.dart

```diff
@@ -0,0 +1,577 @@
+///  * [Step]
+enum StepState {
+  /// A step that is disabled and does to react to taps.
+  disabled,
+  /// A step that is currently inactive, e.g. one that the user did not yet see.
+  inactive,
+  /// A step that is currently active, e.g. one that the user has seen and
+  /// edited.
+  active,
+  /// A step that is currently having an error. e.g. the use has submitted wrong
+  /// input.
+  error
+}
+
+const TextStyle _kStepStyle = const TextStyle(
+  fontSize: 12.0,
+  color: Colors.white
+);
+const TextStyle _kActiveTitleStyle = const TextStyle(
+  fontSize: 14.0,
+  fontWeight: FontWeight.w500,
+  color: Colors.black87
+);
+const TextStyle _kActiveSubtitleStyle = const TextStyle(
+  fontSize: 12.0,
+  color: Colors.black54
+);
+const TextStyle _kDisabledTitleStyle = const TextStyle(
+  fontSize: 14.0,
+  fontWeight: FontWeight.w500,
+  color: Colors.black38
+);
+const TextStyle _kDisabledSubtitleStyle = const TextStyle(
+  fontSize: 12.0,
+  color: Colors.black38
+);
+final TextStyle _kErrorTitleStyle = new TextStyle(
+  fontSize: 14.0,
+  fontWeight: FontWeight.w500,
+  color: Colors.red[500]
+);
+final TextStyle _kErrorSubtitleStyle = new TextStyle(
+  fontSize: 12.0,
+  color: Colors.red[500]
+);
+
+/// A Material step used in [Stepper]. The step can have a title and subtitle,
+/// an icon within its circle, some content and a state that governs its stying.
+///
+/// See also:
+///
+///  * [Stepper]
+///  * <https://material.google.com/components/steppers.html>
+class Step {
+  /// Creates a step.
+  ///
+  /// Arguments [title], [content], and [state] cannot be null.
+  Step({
+    @required this.title,
+    this.subtitle,
+    @required this.content,
+    this.circleChild,
+    this.state: StepState.inactive
+  }) {
+    assert(this.title != null);
+    assert(this.content != null);
+    assert(this.state != null);
+  }
+
+  /// The title of the step. It changes color based on the state.
+  final String title;
+
+  /// The subtitle of the step. It changes color based on the state.
+  ///
+  /// If null, the subtitle is not shown.
+  final String subtitle;
+
+  /// The content of the step that appears below the title and subtitle.
+  ///
+  /// Underneath the content, every step as a 'continue' and 'cancel' button.
+  final Widget content;
+
+  /// The centered child of the step's colored circle.
+  ///
+  /// If null, the circle will be populated with the step's index within the
+  /// stepper.
+  final Widget circleChild;
+
+  /// The state of the step which determines the styling of its componenents
+  /// and whether steps are interactive.
+  final StepState state;
+}
+
+/// A Material stepper widget that keeps track of its current step and offers
+/// customizability and interactivity.
+///
+/// See also:
+///
+///  * [Step]
+///  * <https://material.google.com/components/steppers.html>
+class Stepper extends StatefulWidget {
+  /// Creates a stepper.
+  ///
+  /// Arguments [steps], [axis], and [currentStep] cannot be null.
```

Comment:
Everything except the first phrase of documentation should be in full sentences.